src/screens/AccountNew.js

```diff
@@ -16,191 +16,174 @@
-										seed: '',
-										validBip39Seed: false
-									});
-								}
-							}}
-							network={selectedNetwork}
-							value={address && address}
-						/>
-						<Text style={styles.title}>NAME</Text>
-						<TextInput
-							onChangeText={name => accounts.updateNew({ name })}
-							value={name}
-							placeholder="Enter a new account name"
-						/>
-						{isSubstrate && (
-							<DerivationPathField
-								onChange={({
-									derivationPassword,
-									derivationPath,
-									isDerivationPathValid
-								}) => {
-									this.setState({
+										address: newAddress,
 										derivationPassword,
 										derivationPath,
-										isDerivationPathValid
+										seed: suri,
+										seedPhrase: newSeed,
+										validBip39Seed: isBip39
 									});
-								}}
-								styles={styles}
-							/>
-						)}
-					</View>
-					<View style={styles.bottom}>
-						<Text style={styles.hintText}>
-							Next, you will be asked to backup your account, get a pen and some
-							paper.
-						</Text>
-						<Button
-							buttonStyles={styles.nextStep}
-							title="Next Step"
-							disabled={
-								!validateSeed(seed, validBip39Seed).valid ||
-								!isDerivationPathValid
+								} catch (e) {
+									console.error(e);
+								}
+							} else {
+								// Ethereum account
+								accounts.updateNew({
+									address: newAddress,
+									seed: newSeed,
+									validBip39Seed: isBip39
+								});
 							}
-							onPress={() => {
-								validateSeed(seed, validBip39Seed).valid &&
-									navigation.navigate('AccountBackup', {
-										isNew: true,
-										isWelcome: navigation.getParam('isWelcome')
-									});
-							}}
-						/>
-					</View>
-				</KeyboardScrollView>
+						} else {
+							accounts.updateNew({
+								address: '',
+								seed: '',
+								validBip39Seed: false
+							});
+						}
+					}}
+					network={selectedNetwork}
+					value={address && address}
+				/>
+				<Text style={styles.title}>NAME</Text>
+				<TextInput
+					onChangeText={input => accounts.updateNew({ name: input })}
+					value={name}
+					placeholder="Enter a new account name"
+				/>
+				{isSubstrate && (
+					<DerivationPathField
+						onChange={newDerivationPath => {
+							updateState({
+								derivationPassword: newDerivationPath.derivationPassword,
+								derivationPath: newDerivationPath.derivationPath,
+								isDerivationPathValid: newDerivationPath.isDerivationPathValid
+							});
+						}}
+						styles={styles}
+					/>
+				)}
+				<View style={styles.bottom}>
+					<Text style={styles.hintText}>
+						Next, you will be asked to backup your account, get a pen and some
+						paper.
+					</Text>
+					<Button
+						buttonStyles={styles.nextStep}
+						title="Next Step"
+						disabled={
+							!validateSeed(seed, validBip39Seed).valid ||
+							!isDerivationPathValid
+						}
+						onPress={() => {
+							validateSeed(seed, validBip39Seed).valid &&
```

Comment:
if `validateSeed()` returns false, the button will be disabled anyway, so could remove this check.

Reply:
resolved